de-lib/src/main/java/org/iplantc/de/client/models/UserSettings.java

```diff
@@ -76,326 +42,252 @@ public static UserSettings getInstance() {
         return instance;
     }
 
-    public void setValues(JSONObject obj) {
-        if (obj == null) {
-            return;
-        }
+    public void setUserSettings(UserSetting userSetting) {
+        this.userSetting = userSetting;
+        setUserSettings();
+    }
 
-        setValues(StringQuoter.split(obj.toString()));
+    public UserSetting getUserSetting() {
+        return userSetting;
     }
 
-    public void setValues(Splittable split) {
-        if ((split == null) || (split == Splittable.NULL)) {
-            return;
+    void setUserSettings() {
+        if (userSetting == null) {
+            userSetting = factory.getUserSetting().as();
         }
 
-        if (split.get(EMAIL_ANALYSIS_NOTIFICATION) != null) {
-            setEnableAnalysisEmailNotification(split.get(EMAIL_ANALYSIS_NOTIFICATION).asBoolean());
-        } else {
+        if (userSetting.isEnableAnalysisEmailNotification() == null) {
             setEnableAnalysisEmailNotification(true);
         }
-        if (split.get(EMAIL_IMPORT_NOTIFICATION) != null) {
-            setEnableImportEmailNotification(split.get(EMAIL_IMPORT_NOTIFICATION).asBoolean());
-        } else {
+        if (userSetting.isEnableImportEmailNotification() == null) {
             setEnableImportEmailNotification(true);
         }
 
-        if (split.get(DEFAULT_FILE_SELECTOR_PATH) != null) {
-            setDefaultFileSelectorPath(split.get(DEFAULT_FILE_SELECTOR_PATH).asString());
+        if (Strings.isNullOrEmpty(userSetting.getDefaultFileSelectorPath())) {
+            String homePath = userInfo.getHomePath();
+            setDefaultFileSelectorPath(homePath);
         }
-        if (split.get(REMEMBER_LAST_PATH) != null) {
-            setRememberLastPath(split.get(REMEMBER_LAST_PATH).asBoolean());
-        } else {
+        if (userSetting.isRememberLastPath() == null) {
             setRememberLastPath(true);
         }
 
-        if (split.get(SAVE_SESSION) != null) {
-            setSaveSession(split.get(SAVE_SESSION).asBoolean());
-        } else {
+        if (userSetting.isSaveSession() == null) {
             setSaveSession(true);
         }
 
-        if (split.get(DEFAULT_OUTPUT_FOLDER) != null) {
-            setDefaultOutputFolder(buildFolder(split.get(DEFAULT_OUTPUT_FOLDER)));
+        if (userSetting.getDefaultOutputFolder() == null) {
+            String homePath = userInfo.getHomePath();
+            setDefaultOutputFolder(buildFolder(homePath));
```

Comment:
I'm not sure about using just the home folder as the default output folder. I think this will just dump the `log` and file outputs into the user's home folder. I think it would be safer to use `homePath + "/analyses"`.